crates/mun_memory/src/lib.rs

```diff
@@ -24,6 +24,8 @@ pub trait TypeDesc: Send + Sync {
 pub trait TypeLayout: Send + Sync {
     /// Returns the memory layout of this type.
     fn layout(&self) -> Layout;
+    /// Returns the memory kind of this type, if it is a struct.
+    fn memory_kind(&self) -> Option<abi::StructMemoryKind>;
```

Comment:
Would it be possible to change this to:
`fn is_stack_allocated()` or something like that to avoid a dependency on the abi?

Reply:
Sounds good. Will fix that.